File: javascript/2840.Check-if-Strings-Can-be-Made-Equal-With-Operations-II.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
var checkStrings = function (s1, s2) {
  const even = new Array(26).fill(0);
  const odd = new Array(26).fill(0);

  for (let i = 0; i < s1.length; i++) {
    const idx1 = s1.charCodeAt(i) - 97;
    const idx2 = s2.charCodeAt(i) - 97;

    if (i % 2 === 0) {
      even[idx1] += 1;
      even[idx2] -= 1;
    } else {
      odd[idx1] += 1;
      odd[idx2] -= 1;
    }
  }

  const group1 = even.every((n) => n === 0);
  const group2 = odd.every((n) => n === 0);

  return group1 && group2;
};

File: python/2840.Check-if-Strings-Can-be-Made-Equal-With-Operations-II.py
class Solution:
    def checkStrings(self, s1: str, s2: str) -> bool:
        even = [0] * 26
        odd = [0] * 26

        for i in range(len(s1)):
            idx1 = ord(s1[i]) - ord('a')
            idx2 = ord(s2[i]) - ord('a')

            if i % 2 == 0:
                even[idx1] += 1
                even[idx2] -= 1
            else:
                odd[idx1] += 1
                odd[idx2] -= 1

        return all(n == 0 for n in even) and all(n == 0 for n in odd)